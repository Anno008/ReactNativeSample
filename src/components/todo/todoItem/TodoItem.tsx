import styles from "components/todo/todoItem/TodoItemStyle";
import { ITodoItem } from "models/ITodoItem";
import React from "react";
import { Button, Text, TouchableOpacity } from "react-native";
import { View as AnimatedView } from "react-native-animatable";

interface Props {
    todo: ITodoItem;
    toggleTodo: (id: number) => void;
    deleteTodo: (id: number) => void;
}

export const TodoItem: React.SFC<Props> = (props: Props): JSX.Element => {
    const { todo, toggleTodo, deleteTodo } = props;
    const { finished, name, id } = todo;

    const strikeOutStyle = finished ? styles.strikeout : null;
    const toggleTodoHandler = (): void => toggleTodo(id);
    const deleteTodoHandler = (): void => deleteTodo(id);

    return (
        <AnimatedView animation="fadeInLeft">
            <TouchableOpacity style={styles.container} onPress={toggleTodoHandler}>
                <Text style={[styles.text, strikeOutStyle]}>{name}</Text>
                <Button title="Delete" onPress={deleteTodoHandler} />
            </TouchableOpacity>
        </AnimatedView>
    );
};
