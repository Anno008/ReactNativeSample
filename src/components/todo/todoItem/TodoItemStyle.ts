import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    alignItems: "center",
    alignSelf: "stretch",
    borderColor: "#d6d7da",
    borderRadius: 4,
    borderWidth: 1,
    flexDirection: "row",
    justifyContent: "space-between",
    margin: 5,
    padding: 5
  },
  strikeout: {
    textDecorationLine: "line-through",
    textDecorationStyle: "solid",
  },
  text: {
    fontSize: 30,
    fontWeight: "bold"
  },
});
