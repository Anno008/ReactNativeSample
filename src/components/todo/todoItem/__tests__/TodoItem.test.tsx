import { TodoItem } from "components/todo/todoItem/TodoItem";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import { Button, TouchableOpacity } from "react-native";

describe("TodoItem component test", (): void => {
    // Arrange on top as not to repeat it in every test case
    const mockTodoItem = {
        finished: false,
        id: 1,
        name: "test"
    };
    const mockedProps = {
        deleteTodo: jest.fn(),
        todo: mockTodoItem,
        toggleTodo: jest.fn()
    };

    it("Should match TodoItem snapshot with finished set to false", (): void => {
        const wrapper: ShallowWrapper = shallow(<TodoItem {...mockedProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("Should match TodoItem snapshot with finished set to true", (): void => {
        const finishedTodoProps =  { ...mockedProps, todo: {...mockedProps.todo, finished: true}};

        const wrapper: ShallowWrapper = shallow(<TodoItem {...finishedTodoProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("Should call toggleTodo on press", (): void => {
        const wrapper: ShallowWrapper = shallow(<TodoItem {...mockedProps} />);

        wrapper.find(TouchableOpacity).simulate("press");

        expect(mockedProps.toggleTodo).toBeCalled();
        expect(mockedProps.toggleTodo).toBeCalledWith(mockedProps.todo.id);
    });

    it("Should call deleteTodo on button press", (): void => {
        const wrapper: ShallowWrapper = shallow(<TodoItem {...mockedProps} />);

        wrapper.find(Button).simulate("press");

        expect(mockedProps.deleteTodo).toBeCalled();
        expect(mockedProps.deleteTodo).toBeCalledWith(mockedProps.todo.id);
    });
});
