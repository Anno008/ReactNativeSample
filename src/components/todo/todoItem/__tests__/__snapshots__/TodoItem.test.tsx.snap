// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem component test Should match TodoItem snapshot with finished set to false 1`] = `ShallowWrapper {}`;

exports[`TodoItem component test Should match TodoItem snapshot with finished set to true 1`] = `ShallowWrapper {}`;
